### DISCUSSION TITLE
Different Prediction of Formation Energy for the Potential generated to that of DFT prediction of Formation Energy

### DISCUSSION THREAD

**Discussion by mm21d013** [materialyzeai/matgl#374] [General]
Please refer to the attached image
![Screenshot from 2024-09-26 05-59-20](https://github.com/user-attachments/assets/712ec473-d533-460a-b266-9f5f18c88fc3)

(i) Lets say if I have three structures of Silicon (the image is attached). If I take Fd3̅m1,P6₃/mmc, and C12/m1 space group structures, do some quantum level analysis on it (like some DFT calculations) and use the same as training data for potential development, I get good results.
(ii) If I only take P6₃/mmc, and C12/m1 structures , do DFT calculations and fit in as training data for potential development, the results aren't good.By "results aren't good", I mean...the formation energy predicted via the potential isn't matching with the formation energy predicted by my DFT calculations for those structures. Even though in all this process, the RMSE errors for energy is decreasing with increasing number of epochs...for both the training energy and validation energy

The reason I feel why this happens is because in the (i) case, I have taken the structures which consists of most stable ie...Fd3̅m space group structures alongwith the metastable or the unstable structure(since they have certain energies above the hull plot). However in the (ii) case, I have only taken the metastable or the unstable structure. The stable structure hasn't been part of the training data.

I just wanted to confirm if my inference correct or not ?

Note:I always predict on structures on which I am training in.

**Comment by JiQi535** [comment]
@mm21d013 Hi, could you provide more detailed information about the training sets and test errors in the two cases you discussed? For example, in case 2, where you "only take P6₃/mmc, and C12/m1 structures", what structures are in the training set? How was the formation energy calculated? Is it from relaxation results or just static energy calculation? 

As training error is going down, I suppose static energy calculation is becoming more accurate, while relaxation might still be challenging if training structures do not cover sufficient space.

From my own experience, if we have sufficient sampling of structures related to one local minimum structure, e.g., the relaxed structure of Si with P6₃/mmc space group, the MLIP should be able to get reasonable matching with DFT for that local minimum. But if the sampling is not good enough, it is normal for MLIP to have unreliable extrapolation results. Sufficient sampling usually requires inclusion of equation of state structures, strained structures, relaxation trajectory snapshots, molecular dynamics trajectory snapshots, etc. You can refer to our previous works on MLIP and see how training set is constructed to cover sufficiently large space of structures. [(1,](https://www.nature.com/articles/s41524-024-01227-4) [2,](https://materialsvirtuallab.org/pubs/10.1021_acs.jpca.9b08723.pdf) [3)](https://materialsvirtuallab.org/pubs/10.1016_j.mtphys.2021.100463.pdf)

And I don't think that stable structures are compulsory to be included, though it is a common practice, as augmentation of training set with known stable phases improve the overall reliability of MLIP to the whole system of interest. I'm happy to learn more and see if this understanding is correct.

**Reply by mm21d013** [reply]
@JiQi535 Thanks for that detailed explanation. I demonstrated Si as an example to narrate the problem in the material I am working in. Though I am not disclosing the name of the element/compounds/solid-state solution, what essentially I have covered in those calculations involves strained structures for the element as well as the solid state solution.

In fact, in these calculations, it seems that a mistake has been committed from my end in the DFT calculation. The unstrained structure has non-zero elastic energy (since I took the structure directly from materials project and strained it for my own purpose without any further electronic or ionic relaxation of my unstrained structure). In addition to that the structure itself has energy of 15meV/atom above the hull plot.

However, these structures alongwith their energies were part of training in m3gnet model. Prediction of the formation energy for the same structure should atleast come right. It isn't. In fact the prediction goes absurdly weird when I take a combination of the element with the solid state solution. A cut-off radius of 5angstrom should be good enough looking at the structure, but the prediction of formation energy isn't being good. Even I tweaked the cut-off radius to a larger value but that didn't seem to work.

Similar calculations of energy prediction with MTP potential gives a good result for the same set of structures.

Now, I am a bit confused with the strategy to move from this point. The question that arises are:
(i) Should I re-do the DFT calculations yet again...now making the unstrained structure completely relaxed till it has zero strain energy?
(ii)Should I add an extra list of stable structure in combination to mine (I tried it, the results were atleast much better)?
(iii) I have done certain AIMD calculations over a range of temperatures (from 10K to 300K), should inclusion of that help?
